src/gennylib/include/gennylib/PhaseLoop.hpp

```diff
@@ -103,8 +103,21 @@ class IterationChecker final {
                            phaseContext["SleepBefore"].maybe<TimeSpec>().value_or(TimeSpec{}),
                            phaseContext["SleepAfter"].maybe<TimeSpec>().value_or(TimeSpec{}),
                            phaseContext["GlobalRate"].maybe<RateSpec>()) {
-        const auto rateSpec = phaseContext["GlobalRate"].maybe<RateSpec>();
+        if (!phaseContext.isNop() && !phaseContext["Duration"] && !phaseContext["Repeat"] &&
+            phaseContext["Blocking"].maybe<std::string>() != "None") {
+            std::stringstream msg;
```

Comment:
> Do you think it's more intuitive to set Blocking to False/True instead?

I have a sort of moral objection to booleans since they usually end up wanting to be enums. I could imagine another kind of `Blocking: {ActorsOfType:Crud}` or something in the future. For now all values except `None` are rejected so we have room to expand it in the future.

Reply:
OK that sounds fine. I'm not morally attached to any syntax.

Although, I would very much hope we don't expand on the Blocking syntax. As far as this ticket is concerned, the ideal longer term solution is to use the strict mode check, which can prevent accidental empty phases while being transparent to the user.